inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -1319,6 +1319,69 @@ void GNAGraphCompiler::EltwisePrimitive(InferenceEngine::CNNLayerPtr layer) {
     }
 }
 
+void GNAGraphCompiler::GemmPrimitive(InferenceEngine::CNNLayerPtr layer) {
+    auto quantized = InferenceEngine::getInjectedData<QuantizedLayerParams>(layer);
+
+    IE_ASSERT(!layer->insData.empty());
+    IE_ASSERT(!layer->outData.empty());
+    IE_ASSERT(layer->insData.size() == 2);
+    auto input_1 = layer->insData[0].lock();
+    auto input_2 = layer->insData[1].lock();
+    auto outputs = *layer->outData.begin();
+    auto inputPrecision = quantized ? Precision(Precision::I16) : input_1->getPrecision();
+    uint32_t noOfInputsDivisor = GNALimitations::noOfInputsDivisor;
+
+    auto input_data = HasTo2DReshapeData(layer) ? Get2DReshapedData(input_1, 8) : input_1;
```

Comment:
I suppose, 2d reshape (change of batch size) seems to be possible only for diagonal layer, no fully connected or gemm. Are you sure that it can be applicable for Gemm? Anyway HasTo2DReshapeData(layer) will return false for it.